### PR TITLE
5946 - Fixed pasting from Notion

### PR DESCRIPTION
**Pull request by lzach83**
## Link to Issue
Closes: #5946 

## Description of Changes
- Create an extendible hook `useNotionPaste` which ensures the pasted material is correctly formatted to follow our GH MD standards

## "How We Fixed It"

- Created a hook that will take in the pasted string from Notion and manipulates it to be in the correct format
## Test Plan
- Go to any thread 
- Go to notion.so
- Copy and paste any material and observe

### CODE REVIEW

**Comment by lzach83** [comment]
Converting this back to a draft -- there's a pretty major cursor regression that has been introduced here.

File: packages/commonwealth/client/scripts/views/components/react_quill_editor/react_quill_editor.tsx [hunk 1]
```diff
@@ -121,12 +125,12 @@ const ReactQuillEditor = ({
 
   const handleChange = (value, delta, source, editor) => {
     const newContent = convertTwitterLinksToEmbeds(editor.getContents());
-    console.log('fIRED contentDelta', contentDelta);
     setContentDelta({
       ...newContent,
       ___isMarkdown: true,
     } as SerializableDeltaStatic);
   };
+  //
```
Comment (Israellund) [inline packages/commonwealth/client/scripts/views/components/react_quill_editor/react_quill_editor.tsx:133]: This can be deleted

**Review by Israellund — APPROVED** [review]
LGTM!

**Review by Miaplacidus — APPROVED** [review]
Nice work, Zach! LGTM

**Pull request merged**